sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -429,10 +439,14 @@ public Double getDouble(String propertyName) {
                     T t = (T) ModelBridgeInternal.instantiateJsonSerializable((ObjectNode) n, c);
                     result.add(t);
 
-                } else if (isJsonSerializableWrapper) {
-                    // JsonSerializableWrapper
-                    T t = (T) ModelBridgeInternal.instantiateJsonSerializableWrapper((ObjectNode) n, c);
-                    result.add(t);
+                } else if (containsJsonSerializable) {
+                    try {
+                        T t = c.getDeclaredConstructor(String.class).newInstance(Utils.toJson(Utils.getSimpleObjectMapper(), (ObjectNode) n));
```

Comment:
seems to me this may result in double serialization/deserialization.
can't we just pass ObjectNode to the constructor?

```suggestion
T t = c.getDeclaredConstructor(ObjectNode.class).newInstance((ObjectNode) n));
```

Reply:
Make sense. Changed to use objectNode and added objectNode related constructor which is missing in some of the classes.